core/capabilities/remote/registration/client.go

```diff
@@ -0,0 +1,138 @@
+package registration
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	commoncap "github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+)
+
+type clientRegistration struct {
+	registrationRequest []byte
+}
+
+type registerDispatcher interface {
```

Comment:
You should be able to just use the types.Dispatcher interface. Is there some unexpected circular dependency?